common/src/View/FlyModeHelper.cpp

```diff
@@ -109,79 +111,79 @@ namespace TrenchBroom {
         }
 
         bool FlyModeHelper::keyDown(wxKeyEvent& event) {
-            PreferenceManager& prefs = PreferenceManager::instance();
-            const KeyboardShortcut& forward = prefs.get(Preferences::CameraFlyForward);
-            const KeyboardShortcut& backward = prefs.get(Preferences::CameraFlyBackward);
-            const KeyboardShortcut& left = prefs.get(Preferences::CameraFlyLeft);
-            const KeyboardShortcut& right = prefs.get(Preferences::CameraFlyRight);
-            const KeyboardShortcut& up = prefs.get(Preferences::CameraFlyUp);
-            const KeyboardShortcut& down = prefs.get(Preferences::CameraFlyDown);
+            const KeyboardShortcut& forward = pref(Preferences::CameraFlyForward);
+            const KeyboardShortcut& backward = pref(Preferences::CameraFlyBackward);
+            const KeyboardShortcut& left = pref(Preferences::CameraFlyLeft);
+            const KeyboardShortcut& right = pref(Preferences::CameraFlyRight);
+            const KeyboardShortcut& up = pref(Preferences::CameraFlyUp);
+            const KeyboardShortcut& down = pref(Preferences::CameraFlyDown);
 
             wxCriticalSectionLocker lock(m_critical);
-            
-            if (forward.matches(event)) {
+
+            bool anyMatch = false;
+            if (!m_forward && forward.matchesKeyDown(event)) {
```

Comment:
I think the !m_forward (same with back, etc) part of the condition should be removed so that key repeats are still consumed. Otherwise I get the macOS bell sound when I hold down W,A,S or D.